lib/active_record/pg_enum/postgresql_adapter.rb

```diff
@@ -13,29 +13,46 @@ module PostgreSQLAdapter
       #
       #   { "foo_type" => ["foo", "bar", "baz"] }
       def enum_types
-        res = exec_query(<<-SQL.strip_heredoc, "SCHEMA")
-          SELECT t.typname AS enum_name, array_agg(e.enumlabel ORDER BY e.enumsortorder) AS enum_value
-          FROM pg_type t
-          JOIN pg_enum e ON t.oid = e.enumtypid
-          JOIN pg_catalog.pg_namespace n ON n.oid = t.typnamespace
-          WHERE n.nspname = ANY (current_schemas(false))
-          GROUP BY enum_name;
-        SQL
+        pg_type = Arel::Table.new(:pg_type)
+        pg_enum = Arel::Table.new(:pg_enum)
+        pg_namespace = Arel::Table.new(:pg_namespace)
+        current_schema = Arel::Nodes::NamedFunction.new('current_schemas', [Arel::Nodes::False.new])
+        any_current_schema = Arel::Nodes::NamedFunction.new('ANY', [current_schema])
```

Comment:
So you've replaced perfectly readable SQL with much harder-to-understand Arel code... why?

I'm a big advocate of Arel, I use it all the time in application development. But I intentionally made a design choice with this gem to _not_ use it because I'm targeting a specific database, so the entire reason for using Arel is moot.

Reply:
IMHO the AREL is easier to real, but with that said I took this from my own unpublished postgres enum gem but its really out of scope for the pull request.  Really more than anything I want to have an easier helper AND one that can reflect on the current values in the enum